server/app/repository/ProgramRepository.java

```diff
@@ -258,11 +258,13 @@ public PaginationResult<Application> getApplicationsForAllProgramVersions(
       if (search.matches("^\\d+$")) {
         query = query.eq("id", Integer.parseInt(search));
       } else {
+        Optional<Long> applicantId = getApplicantIdByEmail(search);
```

Comment:
This is going to require the email address to be an exact match. Is that ok? If not, we could join the Applicant table onto the current query and search for partial matches with the rest of the where clauses below.

As written, this also results in an extra query. That's probably fine, but figured I'd call it out.

Reply:
Doesn't `.ieq("submitter_email", search)` require it to be a perfect match? I think a partial match is better, but I'm not sure how to implement that.

I also think the fewer queries the better, but I'm not sure how to do that since email is stored on the `accounts` table. What do you think?

Reply:
we could join the Applicant table onto this query, right?

Reply:
Good call, I'll try that.

Reply:
Argh, I think they won't let me join them because ebean has a rule about [not allowing more than one ](https://ebean.io/docs/query/fetch#rules)`oneToMany` joins. Account is `oneToMany` with Applicant (which [according to this comment](https://github.com/civiform/civiform/blob/4f196c7e83d5fc7fc1c14145161056852128423a/server/app/models/Applicant.java#L28) it actually shouldn't be) and Applicant is `oneToMany` with Applications. Does my reading of this match your understanding?

re: Searching for a partial email match, I tested it out and the current logic for searching my TI email requires it to be a perfect match which seems intentional based on [this test case](https://github.com/civiform/civiform/blob/793e05d7e9e9f0533dd7e68d9925619c822544a4/server/test/repository/ProgramRepositoryTest.java#L291). That said, I think it's a better user experience for email search to mirror name search, so I updated this to search partial strings for applicant emails. I left the current logic that TI emails must be an exact match. What do you think? Is that confusing?

Reply:
Ok cool, I'll update the TI email search to match too. I agree it makes sense for all search to work the same way.

Doing some further reading on the joins, I think the error I was getting ("invalid fetch path") is actually unrelated to the `toMany` joins. But my searching for it led me to this [StackOverflow](https://stackoverflow.com/questions/51882391/ebean-query-operator-fetch-isnt-working-when-i-use-more-than-one-relationship) where they talk about `fetch` doing multiple queries under the hood when you're joining more than one `toMany` relationship.

Reply:
I updated search for both TI and applicant emails to be findable by a partial match.

I paired with @nb1701 a bit on how to join the accounts table, and I'm pretty sure we would have to write some raw sql that makes this much less readable. Since this is on the admin side, I'm opting to keep the extra db hit in favor of keeping the code simpler.

Reply:
Jk I figured out how to do a join with fetch! If at first you don't succeed, write 40 million comments on your own PR about it 😄

Reply:
Amazing! So what was the end result? Is `fetch()` doing multiple queries underneath? It's still easier to read then doing the query separately ourselves.

Reply:
@avaleske see https://ebean.io/docs/query/aggregation#fetch